src/max_key.ts

```diff
@@ -2,17 +2,15 @@
  * A class representation of the BSON MaxKey type.
  */
 export class MaxKey {
-  /**
-   * @ignore
-   */
-  toExtendedJSON() {
+  _bsontype!: 'MaxKey';
```

Comment:
Is this necessary? If so shouldn't it be `{ value: ... }`?

Reply:
I noticed this around a lot, would love to learn more about the difference between this and the `prototype` addition to the class. Could just be `_bsontype = { value: "MaxKey" }` curious why not.

Reply:
We still want it to be a string so we would want it to be `_bsontype = "MaxKey"` but defining it this way makes it an editable property on the MaxKey instance and not on its prototype. Also by defining it with a call to `Object.defineProperty` by default the property descriptor is 
```
{
  value: 'MaxKey',
  writable: false,
  enumerable: false,
  configurable: false
}
```
So it is neither writable, enumerable, nor configurable. In addition to this, and I could be mistaken but the prototype is not editable outside the module so you can't change this once its exported which is good for security.